src/Backend/Modules/Pages/Actions/Add.php

```diff
@@ -553,70 +557,77 @@ private function validateForm(): void
+                // Persist page
+                $allowMove = true;
+                $allowChildren = true;
+                $allowEdit = true;
+                $allowDelete = true;
 
                 if ($this->isGod) {
-                    $page['allow_move'] = in_array(
+                    $allowMove = in_array(
                         'move',
                         (array) $this->form->getField('allow')->getValue(),
                         true
                     );
-                    $page['allow_children'] = in_array(
+                    $allowChildren = in_array(
                         'children',
                         (array) $this->form->getField('allow')->getValue(),
                         true
                     );
-                    $page['allow_edit'] = in_array(
+                    $allowEdit = in_array(
                         'edit',
                         (array) $this->form->getField('allow')->getValue(),
                         true
                     );
-                    $page['allow_delete'] = in_array(
+                    $allowDelete = in_array(
                         'delete',
                         (array) $this->form->getField('allow')->getValue(),
                         true
                     );
                 }
 
-                // set navigation title
-                if ($page['navigation_title'] == '') {
-                    $page['navigation_title'] = $page['title'];
+                $title = $this->form->getField('title')->getValue();
+                $navigationTitle = $this->form->getField('navigation_title')->getValue();
+                if ($navigationTitle === '') {
+                    $navigationTitle = $title;
                 }
 
-                // insert page, store the id, we need it when building the blocks
-                $page['revision_id'] = BackendPagesModel::insert($page);
+                /** @var PageRepository $pageRepository */
+                $pageRepository = BackendModel::get(PageRepository::class);
+                /** @var MetaRepository $metaRepository */
+                $metaRepository = BackendModel::get('fork.repository.meta');
+
+                $meta = $metaRepository->find($this->meta->save());
+
+                $page = new Page(
+                    BackendPagesModel::getMaximumPageId() + 1,
+                    BackendAuthentication::getUser()->getUserId(),
+                    $parentId,
+                    $templateId,
+                    $meta,
+                    BL::getWorkingLanguage(),
+                    $title,
+                    $navigationTitle,
+                    new DateTime(),
+                    BackendPagesModel::getMaximumSequence($parentId) + 1,
+                    $this->form->getField('navigation_title_overwrite')->isChecked(),
+                    $this->form->getField('hidden')->getValue(),
+                    $status,
+                    $parentPage ? 'page' : 'root',
```

Comment:
I would use a value object for this, and then maybe a static constructor that does this check